### PR TITLE
action: release

### PR DESCRIPTION
**Pull request by v1v**
### What

GitHub workflows to run the release steps for:
- merge commits in main
- tag releases

### Test

See https://github.com/elastic/apm-agent-dotnet/actions/runs/4016894734/jobs/6900535051 that was triggered agains this particular PR, as it was configured to listen for events on PRs, it won't be the case anymore

### Follow ups

- Opbeans dotnet automation will be done in the future, that's likely to be implemented within the opbeans-dotnet itself.

### CODE REVIEW

File: .github/workflows/release-main.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: release-main
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
```
Comment (v1v) [inline .github/workflows/release-main.yml:6]:
```suggestion
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          secret/apm-team/ci/elastic-observability-nuget apiKey | REPO_API_KEY ;
+          secret/apm-team/ci/elastic-observability-nuget url | REPO_API_URL
+
+    - name: Release to Nuget
+      run: .ci/linux/deploy.sh ${REPO_API_KEY} ${REPO_API_URL}
+
+    - uses: elastic/apm-pipeline-library/.github/actions/docker-login@current
+      with:
+        registry: docker.elastic.co
+        secret: secret/apm-team/ci/docker-registry/prod
```
Comment (v1v) [inline .github/workflows/release.yml:46]:
```suggestion
        registry: docker.elastic.co
        secret: secret/observability-team/ci/docker-registry/prod
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+name: release
+
+on:
```
Comment (Mpdreamz) [inline .github/workflows/release.yml:3]: Linking conversation about the appropiate trigger here: https://github.com/elastic/ecs-dotnet/commit/b5e88ff2d9dbb495368326028ad4f4ee91ea02e4

**Review by Mpdreamz — APPROVED** [review]

**Comment by Mpdreamz** [comment]
@v1v this LGTM but please wait with pulling while we release 1.20.0 today :smile_cat:

**Comment by v1v** [comment]
@Mpdreamz , @amannocci is leading this migration, I defer to him to decide how to proceed 🙏

**Comment by Mpdreamz** [comment]
`- Build MSBuild - Windows`

Is failing due to restore problems https://status.nuget.org/ will focus on this once we move off Jenkins.

**Pull request merged**